src/schema/v2/me/myCollectionAuctionResults.ts

```diff
@@ -0,0 +1,142 @@
+  args: pageable({
+    sort: AuctionResultSorts,
+    organizations: {
+      type: new GraphQLList(GraphQLString),
+      description: "Filter auction results by organizations",
+    },
+    sizes: {
+      type: new GraphQLList(ArtworkSizes),
+      description: "Filter auction results by Artwork sizes",
+    },
+    categories: {
+      type: new GraphQLList(GraphQLString),
+      description: "Filter auction results by category (medium)",
+    },
+    recordsTrusted: {
+      type: GraphQLBoolean,
+      defaultValue: false,
+      description: "When true, will only return records for allowed artists.",
+    },
+    earliestCreatedYear: {
+      type: GraphQLInt,
+      description: "Filter auction results by earliest created at year",
+    },
+    latestCreatedYear: {
+      type: GraphQLInt,
+      description: "Filter auction results by latest created at year",
+    },
+    allowEmptyCreatedDates: {
+      type: GraphQLBoolean,
+      defaultValue: true,
+      description:
+        "Filter auction results by empty artwork created date values",
+    },
+  }),
+  resolve: async (
+    { id: userId },
+    options,
+    { collectionArtistsLoader, auctionLotsLoader }
+  ) => {
+    try {
+      if (!collectionArtistsLoader || !auctionLotsLoader) return null
+
+      const { body: artists } = await collectionArtistsLoader("my-collection", {
+        user_id: userId,
+        size: 100,
+      })
+
+      const artistIds = artists.map(({ id }) => id)
+
+      if (!artistIds || artistIds.length === 0) {
+        return null
+      }
+
+      const {
+        page,
+        size,
+        offset,
+        sizes,
+        organizations,
+        categories,
+      } = convertConnectionArgsToGravityArgs(options)
+
+      const auctionLotsLoaderArgs = {
+        page,
+        size,
+        artist_ids: artistIds,
+        organizations,
+        categories,
+        earliest_created_year: options.earliestCreatedYear,
+        latest_created_year: options.latestCreatedYear,
+        allow_empty_created_dates: options.allowEmptyCreatedDates,
+        sizes,
+        sort: options.sort,
+      }
+
+      const { total_count, _embedded } = await auctionLotsLoader(
+        auctionLotsLoaderArgs
+      )
+
+      const totalPages = Math.ceil(total_count / size)
+
+      // filter items without artist id
+      const filteredAuctionResults = _embedded.items.filter(
+        (auctionResult) => auctionResult.artist_id
+      )
+
+      // enrich result with artist data
+      const enrichedAuctionResults = filteredAuctionResults.map(
+        (auctionResult) => {
+          const artist = artists.find(
+            ({ _id }) => _id == auctionResult.artist_id
+          )
+
+          auctionResult.artist = artist
+          return auctionResult
+        }
+      )
+
+      return merge(
+        { pageCursors: createPageCursors({ page, size }, total_count) },
+        { totalCount: total_count },
```

Comment:
**thought:** shouldn't this be `filteredAuctionResults.length` since total_count might return items without an artist id (as per your comment in line 101)

Reply:
`total_count` here is the total count of auction results for the given query from `Diffusion`. When we filter out auction results in line 101 it just removes auction results without matching artists from the current page. I can't remember why we needed to filter them out, but it should not brake pagination.

Reply:
ah cool, I got you now. thank you